components/ai_chat/core/common/test_mojom_printers.h

```diff
@@ -8,18 +8,29 @@
 
 #include <string>
 
-#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-forward.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
 
 namespace ai_chat {
 
 namespace mojom {
 
+#define GENERATE_MOJO_PTR_PRINTER(Type)                         \
+  inline void PrintTo(const Type##Ptr& ptr, std::ostream* os) { \
+    PrintTo(*ptr, os);                                          \
+  }
+
 void PrintTo(const AssociatedContent& content, std::ostream* os);
 void PrintTo(const Conversation& conversation, std::ostream* os);
 void PrintTo(const ToolUseEvent& event, std::ostream* os);
 void PrintTo(const ConversationEntryEvent& event, std::ostream* os);
 void PrintTo(const ConversationTurn& event, std::ostream* os);
 
+GENERATE_MOJO_PTR_PRINTER(AssociatedContent)
+GENERATE_MOJO_PTR_PRINTER(Conversation)
+GENERATE_MOJO_PTR_PRINTER(ToolUseEvent)
+GENERATE_MOJO_PTR_PRINTER(ConversationEntryEvent)
+GENERATE_MOJO_PTR_PRINTER(ConversationTurn)
```

Comment:
optional, but seems like we might as well declare both here if we're doing this:

```suggestion
#define DECLARE_MOJO_PRINTER(Type)                              \
  void PrintTo(const Type& ptr, std::ostream* os);              \
  inline void PrintTo(const Type##Ptr& ptr, std::ostream* os) { \
    PrintTo(*ptr, os);                                          \
  }

DECLARE_MOJO_PRINTER(AssociatedContent)
DECLARE_MOJO_PRINTER(Conversation)
DECLARE_MOJO_PRINTER(ToolUseEvent)
DECLARE_MOJO_PRINTER(ConversationEntryEvent)
DECLARE_MOJO_PRINTER(ConversationTurn)
```